src/outputs/short_form_out.py

```diff
@@ -96,12 +92,51 @@ def run_shortform_prep(
         pd.DataFrame: The dataframe prepared for short form output.
     """
 
-    # TODO: Filter for short-forms and CORA statuses [600, 800]
+    # Filter for short-forms
+    df = df.loc[df["formtype"] == "0006"]
 
-    # create new columns and create a 'year' column
+    # Filter for CORA statuses [600, 800]
+    df = df.loc[df["form_status"].isin(["600", "800"])]
+
+    # Create missing columns and create a 'year' column
     df = create_new_cols(df)
 
     # create columns for headcounts for civil and defense
     df = create_headcount_cols(df, round_val)
 
     return df
+
+
+def create_shortform_df(df: pd.DataFrame, schema: str) -> pd.DataFrame:
+    """Creates the shortform dataframe for outputs with
+    the required columns. The naming
+
+    Args:
+        df (pd.DataFrame): Dataframe containing all columns
+        schema (str): Toml schema containing the old and new
+        column names for the outputs
+
+    Returns:
+        (pd.DataFrame): A dataframe consisting of only the
+        required short form output data
+    """
+    # Load schema using pre-built function
+    output_schema = load_schema(schema)
+
+    # Create dict of current and required column names
+    colname_dict = {
+        output_schema[column_nm]["old_name"]: column_nm
+        for column_nm in output_schema.keys()
+    }
+
+    # Create empty dataframe for outputs
+    output_df = pd.DataFrame()
+
+    # Add required columns from schema to output
+    for key in colname_dict.keys():
+        output_df[colname_dict[key]] = df[key]
```

Comment:
I think I can see a couple more ways to do this, one more concisely. Firstly, don't forget the Pandas `DataFrame` constructor has a `columns` parameter. What could you provide at this stage?

Alternatively, if you still want to do it in two stages, consider `.rename` for the second stage which also takes a `columns` argument. You could avoid looping and making iterative changes to a dataframe which is always good! :)

Reply:
I assumed the order of the output columns mattered so i assigned columns according to the schema order. If they don't matter then I can use an intersection and rename it like this in 2 steps:

    # Create subset dataframe with only the required outputs
    output_df = df[df.columns.intersection(colname_dict.keys())]
    
    # Rename columns to match the output specification
    output_df.rename(columns={key :colname_dict[key] for key in colname_dict},inplace = True)

If the order matters, then I can add this third step:
    
    # Rearrange to match user defined output order
    output_df = output_df[colname_dict.values()]